devito/passes/iet/misc.py

```diff
@@ -204,3 +207,39 @@ def remove_redundant_moddims(iet):
     iet = Transformer(subs, nested=True).visit(iet)
 
     return iet
+
+
+def abridge_dim_names(iet):
+    subs = {}
+    mapper = {}
+    tree = retrieve_iteration_tree(iet)
+    for it in tree:
+        indexeds = FindSymbols('indexeds').visit(it.inner)
+        inds = set().union(*[pull_dims(i, flag=False) for i in indexeds])
+        inds = [i for i in inds if any([d.is_Sub for d in i._defines])]
+        inds = [i for i in inds if not i.is_SubIterator]
+        names = [i.root.name for i in inds]
+
+        mapper.update({i: i._rebuild(i.root.name) for i in inds
+                       if i.root not in it.dimensions and names.count(i.root.name) < 2})
+
+    if mapper:
```

Comment:
It would just do a bunch of unnecessary stuff to generate an empty dict and eventually return an unaltered iet. It's just there to avoid inefficiency. Will reverse the return like you said though